iroh-net/src/ticket/blob.rs

```diff
@@ -0,0 +1,189 @@
+    pub fn into_parts(self) -> (NodeAddr, Hash, BlobFormat) {
+        let Ticket { node, hash, format } = self;
+        (node, hash, format)
+    }
+}
+
+impl Serialize for Ticket {
+    fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        if serializer.is_human_readable() {
+            serializer.serialize_str(&self.to_string())
+        } else {
+            let Ticket { node, format, hash } = self;
+            (node, format, hash).serialize(serializer)
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for Ticket {
+    fn deserialize<D: serde::Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        if deserializer.is_human_readable() {
+            let s = String::deserialize(deserializer)?;
+            Self::from_str(&s).map_err(serde::de::Error::custom)
+        } else {
+            let (peer, format, hash) = Deserialize::deserialize(deserializer)?;
+            Self::new(peer, hash, format).map_err(serde::de::Error::custom)
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use iroh_base::base32;
+    use iroh_test::{assert_eq_hex, hexdump::parse_hexdump};
+
+    use crate::key::{PublicKey, SecretKey};
+    use std::net::SocketAddr;
+
+    use super::*;
+
+    fn make_ticket() -> Ticket {
+        let hash = Hash::new(b"hi there");
+        let peer = SecretKey::generate().public();
+        let addr = SocketAddr::from_str("127.0.0.1:1234").unwrap();
+        let derp_url = None;
+        Ticket {
+            hash,
+            node: NodeAddr::from_parts(peer, derp_url, vec![addr]),
+            format: BlobFormat::HashSeq,
+        }
+    }
+
+    #[test]
+    fn test_ticket_postcard() {
+        let ticket = make_ticket();
+        let bytes = postcard::to_stdvec(&ticket).unwrap();
+        let ticket2: Ticket = postcard::from_bytes(&bytes).unwrap();
+        assert_eq!(ticket2, ticket);
+    }
+
+    #[test]
+    fn test_ticket_json() {
+        let ticket = make_ticket();
+        let json = serde_json::to_string(&ticket).unwrap();
+        let ticket2: Ticket = serde_json::from_str(&json).unwrap();
+        assert_eq!(ticket2, ticket);
+    }
+
+    #[test]
+    fn test_ticket_base32() {
+        let hash =
+            Hash::from_str("0b84d358e4c8be6c38626b2182ff575818ba6bd3f4b90464994be14cb354a072")
+                .unwrap();
+        let node_id = PublicKey::from_bytes(
+            &<[u8; 32]>::try_from(
+                hex::decode("ae58ff8833241ac82d6ff7611046ed67b5072d142c588d0063e942d9a75502b6")
+                    .unwrap(),
+            )
+            .unwrap(),
+        )
+        .unwrap();
+
+        let ticket = Ticket {
+            node: NodeAddr::from_parts(node_id, None, vec![]),
+            format: BlobFormat::Raw,
+            hash,
+        };
+        let base32 = base32::parse_vec(ticket.to_string().strip_prefix("blob").unwrap()).unwrap();
+        let expected = parse_hexdump("
+            00 # discriminator for variant 0
```

Comment:
nice to include a test for this!

Reply:
Without this you could easily break things by moving fields around in the ticket.